api/foxpass_whitelist_sync.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+
+"""
+This script requires the external libraries from requests
+pip install requests
+
+To run:
+python foxpass_whitelist_sync.py --api-key <api_key> --hostname <DDNS_hostname> --whitelist-name <whitelist_name>
+"""
+import argparse
+import json
+import socket
+import sys
+
+import requests
+
+URL = 'https://api.foxpass.com/v1/'
+ENDPOINT = 'whitelist_ips/'
+API = URL + ENDPOINT
+
+
+def main():
+    args = get_args()
+    header = {'Authorization': 'Token ' + args.api_key}
+    whitelist_ip = get_whitelist_ip(header, args.whitelist_name)
+    target_ip = socket.gethostbyname(args.hostname)
+    if whitelist_ip != target_ip:
+        clear_whitelist(header, args.whitelist_name)
+        set_whitelist_ip(header, args.whitelist_name, target_ip)
```

Comment:
this is a dangerous way to accomplish this task. if something goes wrong with `set_whitelist`, then the list will be empty until the next time this script runs.

Better is to get the set of current whitelisted IPs, compare to the set of desired IPs, and add/delete as needed.

Reply:
Comments added on destructive nature of the script.